### PR TITLE
Use "optimization_target" and "optimization_direction" parameters in the optimizer config

### PR DESCRIPTION
**Pull request by motus**

### CODE REVIEW

File: mlos_bench/mlos_bench/tests/config/schemas/optimizers/test-cases/bad/invalid/bad_space_adapter_type.jsonc [hunk 1]
```diff
@@ -2,7 +2,7 @@
     "class": "mlos_bench.optimizers.mlos_core_optimizer.MlosCoreOptimizer",
 
     "config": {
-        "minimize": "score",
+        "optimization_target": "score",
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/tests/config/schemas/optimizers/test-cases/bad/invalid/bad_space_adapter_type.jsonc:5]: Can you add a test where "optimization_direction" is neither "min" nor "max" so we make sure it gets rejected?

File: mlos_bench/mlos_bench/config/schemas/optimizers/optimizer-schema.json [hunk 2]
```diff
@@ -12,17 +12,14 @@
...
             "properties": {
-                "minimize": {
-                    "description": "The name of the metric to minimize.",
-                    "$comment": "In oneOf spec below require one of 'minimize' or 'maximize'.",
-                    "type": "string",
-                    "default": "score"
-                },
-                "maximize": {
-                    "description": "The name of the metric to maximize.",
-                    "$comment": "In oneOf spec below require one of 'minimize' or 'maximize'.",
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/config/schemas/optimizers/optimizer-schema.json:23]: Did you remove the "oneOf" rule being referenced here?

**Review by bpkroth — APPROVED** [review]
Looks pretty good to me, just needs that one extra test I think.

File: mlos_bench/mlos_bench/tests/config/schemas/optimizers/test-cases/bad/invalid/bad_space_adapter_type.jsonc (hunk 1 above)
Comment (motus) [reply mlos_bench/mlos_bench/tests/config/schemas/optimizers/test-cases/bad/invalid/bad_space_adapter_type.jsonc:5]: will do!

File: mlos_bench/mlos_bench/config/schemas/optimizers/optimizer-schema.json (hunk 2 above)
Comment (motus) [reply mlos_bench/mlos_bench/config/schemas/optimizers/optimizer-schema.json:23]: good catch! I'll check it now

**Pull request merged**